File: schemas/email-campaign-marketing/drizzle/contact_tag_assignments.ts
// contact_tag_assignments: Many-to-many join between contacts and tags.
// See README.md for full design rationale.

import { pgTable, uuid, timestamp, index, unique } from "drizzle-orm/pg-core";
import { contacts } from "./contacts";
import { tags } from "./tags";

export const contactTagAssignments = pgTable(
  "contact_tag_assignments",
  {
    id: uuid("id").primaryKey().defaultRandom(),
    contactId: uuid("contact_id").notNull().references(() => contacts.id, { onDelete: "cascade" }),
    tagId: uuid("tag_id").notNull().references(() => tags.id, { onDelete: "cascade" }),
    createdAt: timestamp("created_at", { withTimezone: true }).notNull().defaultNow(),
  },
  (table) => [
    unique("uq_contact_tag_assignments_contact_tag").on(table.contactId, table.tagId),
    index("idx_contact_tag_assignments_tag_id").on(table.tagId),
  ],
);

File: schemas/email-campaign-marketing/spacetimedb/contact_tag_assignments.rs
// contact_tag_assignments: Junction table linking contacts to tags.
// See README.md for full design rationale.

use spacetimedb::Timestamp;

#[spacetimedb::table(name = contact_tag_assignments, public)]
pub struct ContactTagAssignment {
    #[primary_key]
    pub id: String, // UUID

    pub contact_id: String, // UUID, FK → contacts.id (cascade delete)

    #[index(btree)]
    pub tag_id: String, // UUID, FK → tags.id (cascade delete)
    // Composite unique: (contact_id, tag_id) — enforce in reducer logic

    pub created_at: Timestamp,
}
